---
title: "Moth Evolution"
author: "M.A. Hagadorn, A.L. Springer, T. Saley, A. Magby, and A. Rhodes"
date: "September 12, 2017"
output:
  pdf_document:
    fig_caption: yes
    keep_tex: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE)
```

```{r datadownload, include=FALSE, size="footnotesize"}
mothdata <- read.csv("https://raw.githubusercontent.com/mahagadorn/EvolutionaryGenomics/master/MothEvolution/data_medionigra.csv", sep=",", as.is=TRUE)
```
#1)
##*METHODS*
###Simulation Construction and Application
We used a $\beta$-distribution to generate a random sample of population sizes for each simulation (see `pop.vec`). We chose a beta distribution because it cannot produce values of zero or below (effectively preventing our simulated population from going extinct), and because the skew of the distribution better matched the skew towards lower values seen in a histogram of our actual data. The parameters of our beta distribution ($\alpha$ = 1.2 and $\beta$ = 17) were chosen such that the arithmetic mean and harmonic mean of a random sample generated from our distribution matched the arithmetic and harmonic mean of our actual population data (roughly $\mu$ = 1000 and H = 260) as closely as possible. We then scaled our random population size vector (`pop.vec`) such that the harmonic mean, an estimate of Ne under fluctuating population size, of each simulated population vector would match the value chosen by the user. 

The code chunk below includes our equation for calculating effective population size (*Ne*), as well as the model used for generating our simulations.

```{r binomial, include=TRUE}
Ne <- length(mothdata[,2])/sum(1/mothdata[,2])
#harmonic mean estimate for Ne using actual data


model <- function(data, numb.sim, time, Ne){
  df <- matrix(NA, nrow = time, ncol = numb.sim)
  for(j in 1:numb.sim){
    allele.freq.seed <- data[1,3]/data[1,2]
    #initial allele frequency (to be replaced each)
    pop.vec <- 10000*rbeta(time, 1.2, 17) 
    #vector of population sizes for each year of the simulation
    scalar <- (Ne*(sum(1/pop.vec)))/(time) 
    #calculates scalar necessary to make harmonic mean of pop.vec equal to user's choice in Ne
    pop.vec.fit <- round(0.93*scalar*pop.vec, 0) 
    #altered pop.vec with harmonic mean equal to user's choice in Ne
    for(i in 1:length(pop.vec.fit)){
        df[i,j] <- allele.freq.seed 
        #fills data frame
        allele.sim <- rbinom(n=1, pop.vec.fit[i], allele.freq.seed) 
        #Generate next gen's allele count
        if(pop.vec[i] == 0){
           allele.freq.seed <- 0
        }
        allele.freq.seed <- allele.sim/pop.vec.fit[i] 
        #replace allele frequency seed with next gen's number
    }
  }
  return(df)
}
```


###Allele Frequency Change
Using the population size vector generated above, we simulated allele frequency change over time by first seeding the simulation with the initial frequency of the *medionigra* allele from the actual data (0.111), then generating the allele count for year `i + 1` by drawing a sample from a binomial distribution with parameters *N* = population size that year (i.e. the population size generated by `pop.vec.fit` for year `i`) and *p* = allele frequency at year `i`.

We ran the model above under four different scenarios: *Ne* = 50, 263, 500, and 1000. For each scenario, the model was for 1000 iterations, with each iteration being run for a time period of 60 years (1940-1999). The effective population sizes of 50, 500, 1000, were chosen to represent the allele frequency change one might expect in a small, medium, and large population respectively, and the effective population of 263 was chosen to reflect our estimate of *Ne* (the harmonic mean) seen in the observed data.

```{r simulations, include=FALSE}
sims.50 <- model(mothdata, 1000, 60, 50)
sims.263 <- model(mothdata, 1000, 60, Ne)
sims.500 <- model(mothdata, 1000, 60, 500)
sims.1000 <- model(mothdata, 1000, 60, 1000)
```




###*Simulated Data*: Calculating Net Allele Frequency Change and Variance
To evaluate differences among the four scenarios we modeled (*Ne* = 50, 263, 500, 1000), we calculated both net change in allele frequency and variance in in allele frequency for each simulation using the code below. Thus, for each scenario we calculated two values, net $\Delta$*medionigra* and variance in $\Delta$*medionigra*, for each of the 1000 simulations within each of the four models..

```{r netchange, echo=TRUE}
netAl.change <- function(x){
  vec <- numeric(length(x)-1)
    for(i in seq_along(vec)){
      vec[i] <- x[i+1] - x[i]
      #take the difference between the year after i and i
    }
    sumdiff <- sum(vec)
  return(sumdiff)
}
``` 



```{r calculateandcbind, include=FALSE}
integrate <- function(x){
  persim_Alchange <- apply(x, 2, netAl.change)
  #this gives you allele frequency change over time PER simulation!!
  variance <- apply(x, 2, var)
  results <- cbind(persim_Alchange, variance)
  return(results)
}

final.50<-integrate(sims.50)
final.263<-integrate(sims.263)
final.500<-integrate(sims.500)
final.1000<-integrate(sims.1000)
```


###*Observed Dat*a: Calculating Allele Frequency Change and Variance
To assess where the observed *medionegra* allele frequency changes and variance fall in relation to simulated data set, we calculated these values using the following function:
```{r observed}
allelefreq_Ne <- function(x, Ne){
  ObsAfreqchange_Ne <- x[,3]/Ne
  return(cbind(mothdata, Ne,ObsAfreqchange_Ne))
}
```

```{r observedcompile, include=FALSE}
modified.moth <- allelefreq_Ne(mothdata, Ne)
obs.changeAl <- netAl.change(modified.moth[,5])
obs.var <- var(modified.moth[,5])

ObservedStats <- c(obs.changeAl, obs.var)
```

#*RESULTS*
```{r plot.50, echo=FALSE, fig.height=4, fig.width=7, fig.cap= "Plot of variance in *medionigra* allele frequencies against net change in allele frequencies over 60 generations when *Ne*=50.  Individual data points (blue circles) represent the calculated values from a single simulation.  The red X indicates the variance in allele frequency as a function of the net change in allele frequency for the observed data."}
plot<-plot(final.50, type='p', col='blue', pch=20, ylim = c(0,.25), xlab="Total Allele Frequency Change", ylab="Variance of Allele Frequencies", main=expression(italic(Ne)~50~per~Simulation~italic(medionigra)~Changes~Related~to~Variance))
  axis(side=1,at=seq(-.3,1.1,.2),tcl=0.2)
  points(ObservedStats[1],ObservedStats[2], col='red', pch=4)
```


```{r plot.263, echo=FALSE,fig.height=4, fig.width=7, fig.cap= "Plot of variance in *medionigra* allele frequencies against net change in allele frequencies over 60 generations when *Ne*=263.  Individual data points (blue circles) represent the calculated values from a single simulation.  The red X indicates the variance in allele frequency as a function of the net change in allele frequency for the observed data."}
plot<-plot(final.263, type='p', col='blue', pch=20, ylim = c(0,.25), xlab="Total Allele Frequency Change", ylab="Variance of Allele Frequencies", main=expression(italic(Ne)~263~per~Simulation~italic(medionigra)~Changes~Related~to~Variance))
  axis(side=1,at=seq(-.3,1.1,.2),tcl=0.2)
  points(ObservedStats[1],ObservedStats[2], col='red', pch=4)
```


```{r plot.500, echo=FALSE,fig.height=4, fig.width=7, fig.cap= "Plot of variance in *medionigra* allele frequencies against net change in allele frequencies over 60 generations when *Ne*=500.  Individual data points (blue circles) represent the calculated values from a single simulation.  The red X indicates the variance in allele frequency as a function of the net change in allele frequency for the observed data."}
plot<-plot(final.500, type='p', col='blue', pch=20, ylim = c(0,.25), xlab="Total Allele Frequency Change", ylab="Variance of Allele Frequencies", main=expression(italic(Ne)~500~per~Simulation~italic(medionigra)~Changes~Related~to~Variance))
  axis(side=1,at=seq(-.3,1.1,.2),tcl=0.2)
  points(ObservedStats[1],ObservedStats[2], col='red', pch=4)
```


```{r plot.1000, echo=FALSE,fig.height=4, fig.width=7, fig.cap= "Plot of variance in *medionigra* allele frequencies against net change in allele frequencies over 60 generations when *Ne*=1000.  Individual data points (blue circles) represent the calculated values from a single simulation.  The red X indicates the variance in allele frequency as a function of the net change in allele frequency for the observed data."}
plot<-plot(final.1000, type='p', col='blue', pch=20, ylim = c(0,.2), xlab="Total Allele Frequency Change", ylab="Variance of Allele Frequencies", main=expression(italic(Ne)~1000~per~Simulation~italic(medionigra)~Changes~Related~to~Variance))
  axis(side=1,at=seq(-.3,1.1,.2),tcl=0.2)
  points(ObservedStats[1],ObservedStats[2], col='red', pch=4)
```



```{r Quantiles, include=FALSE}
quantiles.fun <- function(x){
  time <- seq(1940, 1999, 1)  #Accounting for the 60 years of data
  sim.mean <- apply(x, 1, mean)
  p05 <- apply(x, 1, quantile, prob=0.05)
  p25 <- apply(x, 1, quantile, prob=0.25)
  p75 <- apply(x, 1, quantile, prob=0.75)
  p95 <- apply(x, 1, quantile, prob=0.95)
  min <- apply(x, 1, min)
  max <- apply(x, 1, max)
  final <- cbind(time,sim.mean,min,p05,p25,p75,p95,max)
  return(final)
}

results_Ne50 <- quantiles.fun(sims.50)
results_Ne263 <- quantiles.fun(sims.263)
results_Ne500 <- quantiles.fun(sims.500)
results_Ne1000 <- quantiles.fun(sims.1000)

transparent_col <- function(color, transparency = 50) {  #color=name of color; name=name of color
#Get standard RGB values for named color
  rgb.val <- col2rgb(color)
#Make new color using input color as base and alpha set by transparency
  transparent_col <- rgb(rgb.val[1], rgb.val[2], rgb.val[3],
               max = 255,
               alpha = (100-transparency)*255/100)
}

col1 <- transparent_col("gray87", .000001)
col2 <- transparent_col("gray91", .000001)
```

```{r Ne50, echo=FALSE,fig.height=5.5, fig.width=7, fig.cap= "Frequency over time for the observed data (red) against the allele frequency over time generated by our simulations where *Ne* = 50. The blue line represents mean allele frequency change generated by the model, while the black lines represent the minimum and maximum changes in allele frequency. The light grey region represents the boundaries within which 90% of the data from our model fell (i.e. the 5th and 95th percentile), and the dark grey region represents the boundaries within which 50% of the data from our model fell (i.e. the 25th and 75th percentiles)."}
Ne50.plot<-plot(results_Ne50[,2]~results_Ne50[,1], ylim=c(0,1), type='l', col='blue', xlab='Years', ylab='Allele Frequency', main=expression(~Simulated~vs~Observed~italic(medionigra)~Frequency~Change~For~italic(Ne50)))
                lines(results_Ne50[,2]~results_Ne50[,1], col='black')
                lines(results_Ne50[,3]~results_Ne50[,1], col='black')
                lines(results_Ne50[,4]~results_Ne50[,1], col=col2)
                lines(results_Ne50[,5]~results_Ne50[,1], col='gray')
                lines(results_Ne50[,6]~results_Ne50[,1], col='gray')
                lines(results_Ne50[,7]~results_Ne50[,1], col=col2)
                lines(results_Ne50[,8]~results_Ne50[,1], col='black')
                polygon(c(results_Ne50[,1],rev(results_Ne50[,1])), c(results_Ne50[,4],rev(results_Ne50[,7])), col=col2, border=NA)  #used to fill the area you want to be transparent
                polygon(c(results_Ne50[,1],rev(results_Ne50[,1])), c(results_Ne50[,5],rev(results_Ne50[,6])), col=col1, border=NA)  #used to fill the area you want to be transparent
                lines(modified.moth[,5]~modified.moth[,1], type='b', col='red')
                lines(results_Ne50[,2]~results_Ne50[,1], type='l', col='blue')
                lines(results_Ne50[,3]~results_Ne50[,1], col='black')
                lines(results_Ne50[,8]~results_Ne50[,1], col='black')
```

```{r Ne263, echo=FALSE,fig.height=5.5, fig.width=7, fig.cap= "Frequency over time for the observed data (red) against the allele frequency over time generated by our simulations where *Ne* = 263. The blue line represents mean allele frequency change generated by the model, while the black lines represent the minimum and maximum changes in allele frequency. The light grey region represents the boundaries within which 90% of the data from our model fell (i.e. the 5th and 95th percentile), and the dark grey region represents the boundaries within which 50% of the data from our model fell (i.e. the 25th and 75th percentiles)."}
Ne263.plot<-plot(results_Ne263[,2]~results_Ne263[,1], ylim=c(0,1), type='l', col='blue', xlab='Years', ylab='Allele Frequency', main=expression(~Simulated~vs~Observed~italic(medionigra)~Frequency~Change~For~italic(Ne263)))
                lines(results_Ne263[,2]~results_Ne263[,1], col='black')
                lines(results_Ne263[,3]~results_Ne263[,1], col='black')
                lines(results_Ne263[,4]~results_Ne263[,1], col=col2)
                lines(results_Ne263[,5]~results_Ne263[,1], col='gray')
                lines(results_Ne263[,6]~results_Ne263[,1], col='gray')
                lines(results_Ne263[,7]~results_Ne263[,1], col=col2)
                lines(results_Ne263[,8]~results_Ne263[,1], col='black')
                polygon(c(results_Ne263[,1],rev(results_Ne263[,1])), c(results_Ne263[,4],rev(results_Ne263[,7])), col=col2, border=NA)  #used to fill the area you want to be transparent
                polygon(c(results_Ne263[,1],rev(results_Ne263[,1])), c(results_Ne263[,5],rev(results_Ne263[,6])), col=col1, border=NA)  #used to fill the area you want to be transparent
                lines(modified.moth[,5]~modified.moth[,1], type='b', col='red')
                lines(results_Ne263[,2]~results_Ne263[,1], type='l', col='blue')
                lines(results_Ne263[,3]~results_Ne263[,1], col='black')
                lines(results_Ne263[,8]~results_Ne263[,1], col='black')
```


```{r Ne500, echo=FALSE,fig.height=5.5, fig.width=7, fig.cap= "Frequency over time for the observed data (red) against the allele frequency over time generated by our simulations where *Ne* = 500. The blue line represents mean allele frequency change generated by the model, while the black lines represent the minimum and maximum changes in allele frequency. The light grey region represents the boundaries within which 90% of the data from our model fell (i.e. the 5th and 95th percentile), and the dark grey region represents the boundaries within which 50% of the data from our model fell (i.e. the 25th and 75th percentiles)."}
Ne500.plot<-plot(results_Ne500[,2]~results_Ne500[,1], ylim=c(0,1), type='l', col='blue', xlab='Years', ylab='Allele Frequency', main=expression(~Simulated~vs~Observed~italic(medionigra)~Frequency~Change~For~italic(Ne500)))
                lines(results_Ne500[,2]~results_Ne500[,1], col='black')
                lines(results_Ne500[,3]~results_Ne500[,1], col='black')
                lines(results_Ne500[,4]~results_Ne500[,1], col=col2)
                lines(results_Ne500[,5]~results_Ne500[,1], col='gray')
                lines(results_Ne500[,6]~results_Ne500[,1], col='gray')
                lines(results_Ne500[,7]~results_Ne500[,1], col=col2)
                lines(results_Ne500[,8]~results_Ne500[,1], col='black')
                polygon(c(results_Ne500[,1],rev(results_Ne500[,1])), c(results_Ne500[,4],rev(results_Ne500[,7])), col=col2, border=NA)  #used to fill the area you want to be transparent
                polygon(c(results_Ne500[,1],rev(results_Ne500[,1])), c(results_Ne500[,5],rev(results_Ne500[,6])), col=col1, border=NA)  #used to fill the area you want to be transparent
                lines(modified.moth[,5]~modified.moth[,1], type='b', col='red')
                lines(results_Ne500[,2]~results_Ne500[,1], type='l', col='blue')
                lines(results_Ne500[,3]~results_Ne500[,1], col='black')
                lines(results_Ne500[,8]~results_Ne500[,1], col='black')
```


```{r Ne1000, echo=FALSE,fig.height=5.5, fig.width=7, fig.cap= "Frequency over time for the observed data (red) against the allele frequency over time generated by our simulations where *Ne* = 1000. The blue line represents mean allele frequency change generated by the model, while the black lines represent the minimum and maximum changes in allele frequency. The light grey region represents the boundaries within which 90% of the data from our model fell (i.e. the 5th and 95th percentile), and the dark grey region represents the boundaries within which 50% of the data from our model fell (i.e. the 25th and 75th percentiles)."}
Ne1000.plot<-plot(results_Ne1000[,2]~results_Ne1000[,1], ylim=c(0,1), type='l', col='blue', xlab='Years', ylab='Allele Frequency', main=expression(~Simulated~vs~Observed~italic(medionigra)~Frequency~Change~For~italic(Ne1000)))
                lines(results_Ne1000[,2]~results_Ne1000[,1], col='black')
                lines(results_Ne1000[,3]~results_Ne1000[,1], col='black')
                lines(results_Ne1000[,4]~results_Ne1000[,1], col=col2)
                lines(results_Ne1000[,5]~results_Ne1000[,1], col='gray')
                lines(results_Ne1000[,6]~results_Ne1000[,1], col='gray')
                lines(results_Ne1000[,7]~results_Ne1000[,1], col=col2)
                lines(results_Ne1000[,8]~results_Ne1000[,1], col='black')
                polygon(c(results_Ne1000[,1],rev(results_Ne1000[,1])), c(results_Ne1000[,4],rev(results_Ne1000[,7])), col=col2, border=NA)  #used to fill the area you want to be transparent
                polygon(c(results_Ne1000[,1],rev(results_Ne1000[,1])), c(results_Ne1000[,5],rev(results_Ne1000[,6])), col=col1, border=NA)  #used to fill the area you want to be transparent
                lines(modified.moth[,5]~modified.moth[,1], type='b', col='red')
                lines(results_Ne1000[,2]~results_Ne1000[,1], type='l', col='blue')
                lines(results_Ne1000[,3]~results_Ne1000[,1], col='black')
                lines(results_Ne1000[,8]~results_Ne500[,1], col='black')
```

Our plots for allele frequency change vs. variance (Fig.2-4) for the simulations where *Ne* = 263, 500, and 1000 showed that the net change in allele frequency for the observed data data fall well within the range of values produced by the model, but the variance was greater the range produced by the model. Similarly, in our plots of allele frequency change over time, the change in *medionigra* in the observed data in many of the years between 1940 and 1960 were above the 95th percentile for change generated by our model, suggesting that drift alone cannot account for the pattern seen if *Ne* is 263 or greater (Fig.6-8). This disparity disappears in both simulations where *Ne* = 50 (Fig. 1 and Fig. 5), suggesting that if the actual *Ne* the moth population is far lower than we estimated (thus increasing stochasticity and drift), then drift alone could explain the pattern seen in the observed data. 

One major limitation of our model is that it does not account for other demographic features of the moth population that could be altering *Ne*. The harmonic mean is a good estimator for *Ne* when fluctuating population size is the major variable influencing *Ne*, but if other factors (i.e. assortative mating, skewed sex ratios, unequal family sizes, etc.) also play a non-trivial role in our moth population, then the harmonic mean will be an overestimate of *Ne*. Thus, it is possible that in reality the *Ne* of the moth population is actually low enough that drift alone could explain the observed changes in allele frequency. However, after 1960, the frequency of the *medionigra* allele in the observed data remains consistently low, which does not fit the expected behavior of a population with a very small *Ne*. At an *Ne* of 50, stochasticity in sampling is so high that the probability of the *medionigra* allele being driven to either fixation or extinction over time is far higher than in a comparable population with a large *Ne*. Thus, even though a small *Ne* could account for the observed pattern, it seems probable that other factors, such as fluctuating selection, are influencing the frequency for the *medionigra* allele over time in addition to drift. 

It is also possible that the lower variance seen in our simulations is an artifact of our beta distribution producing lower fluctuations in population size over time than seen in the actual population. This is possible even though we standardized the harmonic mean to correspond with the harmonic mean of the actual moth data as the same harmonic mean can be produced from a set of numbers with a lower mean and lower variance as a set of numbers with a higher mean and higher variance.




#5)
The *medionigra* time-series data set in the moth *Panaxia dominula* has been analyzed by several different groups throughout the years in an attempt to find a model that explains the changes in allelic frequency in this population. Fisher and Ford (1947) analyzed the data first, which prompted a response by Wright (1948). Over 66 years later, Foll *et al.* (2014) attempted to find the most efficient method to look at effective population sizes and selection in time-series data. We looked at a data set spanning 60 years to see if genetic drift alone could explain the change in allelic frequency observed over time. Based on our results, genetic drift does not likely explain all the observed allelic changes over the time period.

Fisher and Ford (1947) stated that selection is the only mechanism by which evolution will occur because population sizes in natural populations are never low enough that genetic drift or “chance fluctuations in gene-ratios” (Fisher and Ford 1947, pp 173) plays no role. As an example, they compared the frequency of a recessive allele (that produced an easily recognizable phenotypic trait) over a time series (six years) within an isolated population of *P. dominula*, located near Oxford in Cothill Fen, England.  In response to this assertion, Wright (1948) responded that “The fluctuations of some genes are undoubtedly governed largely by violently shifting conditions of selection, but for others in the same populations accidents of sampling should be much more important, and for still others both may play significant roles (Wright 1948, pp 291).”  Wright believed that genetic drift was most likely to have a large effect on small isolated populations.

Foll *et al.* (2014), used an expanded dataset derived from the same population of *P. dominula* _collected over sixty (rather than six) years_ to compare the posterior distribution of the *medionigra* (recessive) allele for populations with *Ne* values of 1000 and 100. They found that genetic drift could explain the observed patterns in smaller populations.

Our model simulated the changes in *medionigra* frequencies through time due to drift, while incorporating varying *Ne* parameters. The results revealed that multiple observed data points lay outside of the range of our simulation (Fig. 6).  This suggests that genetic drift alone is not the only evolutionary mechanism influencing changes in *medionigra* allelic frequency in the  *P. dominula* population. Our results contrast with the results of Foll *et al.* (2014), as we found that even with an effective population as small as *Ne* = 50, there were changes that could not be explained by drift alone. Simulations run with *Ne* = 500 and 1000 showed similar, but progressively stronger deviations from the predicted pattern (Fig. 7 and 8, respectively). However, these deviations from the predicted range of allele frequencies were much smaller and less frequent in populations with the smallest effective population size (*Ne* = 50).

Despite later evidence to the contrary (Foll *et al.* 2014), the likely relative advantageous effect of the *medionigra* allele was, ironically,  agreed upon by Fisher and Ford (1947) and Wright (1948).  Fisher and Ford (1947) stated “It is quite probable...that in this colony the heterozygotes are at an advantage compared with either homozygote (pp 151).”  Wright (1947) conceded “It seems very probable that the heterozygotes actually have a slight net advantage over both homozygotes, as suggested by [Fisher and Ford 1947].”  However, Foll *et al*. (2014) found that a fully recessive lethal model is the best explanation for the patterns hypothesized by Mathieson and McVean (2013).  Their joint posterior distribution model indicates that *medionigra* is a deleterious allele, which is in agreement with previous studies (Sheppard and Cook, 1962).  However, Foll *et al.* (2014) also acknowledge that, “ ...in this case, once the allele frequency is low enough such that heterozygotes almost never occur, it behaves like a neutral model (pp 96).”  Despite his belief that the *medionigra* allele has a net advantage, Wright (1948) observed that there was a large amount of variability in the potential selection coefficient of the allele among sampling seasons.  He stated, “A standard deviation of .22 means that in the course of half a century the selective value of *medionigra* heterozygotes would vary between semi-lethality (or semisterility) (s = -.50) to a 50% advantage over type (s = +.50) (Wright 1948, pp 283).” Given that we never estimated the selection coefficient (*s*), this question of the relative advantage or disadvantage of the *medionigra* allele was not addressed by our group.   It would, however, be interesting to see the posterior distribution for *s* and dominance (*h*) at the true estimated effective population size (*Ne* = 263). 


\newpage
#References Cited

Fisher, R.A., and E.B. Ford (1947).  The spread of a gene in natural conditions in a colony of the moth Panaxia dominula L.  Heredity 1:143-174.

Foll, M., H. Shim, and J.D. Jensen (2014).  WFABC: a Wright-Fisher ABC-based approach for inferring effective population sizes and selection coefficients from time-sampled data.  Molecular Ecology Resources 15:87-98.

Mathieson, I., and G. McVean (2013).  Estimating selection coefficients in spatially structured populations from time series data of allele frequencies.  GEnetics 193:973-984.

Sheppard, P.M., and L.M. Cook (1962).  The manifold effects of the Medionigra gene of the moth Panaxia dominula and the maintenance of a polymorphism.  Heredity 17:415-426

Wright, S. (1948).  On the roles of directed and random changes in gene frequency in the genetics of populations.  Evolution 2:279-294.



